iothub/device/src/Exceptions/IotHubClientErrorCode.cs

```diff
@@ -125,5 +125,10 @@ public enum IotHubClientErrorCode
         /// guide for more information.
         /// </remarks>
         Unauthorized,
+
+        /// <summary>
+        /// The request failed because of TLS authentication error.
+        /// </summary>
+        AuthenticationError,
```

Comment:
I'd consider renaming this ```TlsAuthenticationError``` to be more specific. Otherwise, at a glance, I would be confused why we'd have both ```Unauthorized``` and ```AuthenticationError```